src/compute-client/src/plan/top_k.rs

```diff
@@ -85,6 +87,30 @@ impl TopKPlan {
             })
         }
     }
+    /// Informs the plan that it will apply to a single time, and therefore
+    /// may upgrade from a basic topk plan to a monotonic plan with mandatory
+    /// consolidation.
+    pub fn flag_snapshot(&mut self) {
+        if let TopKPlan::Basic(plan) = self {
```

Comment:
Just curious: in what case will we end up with a `MonotonicTop1Plan` / `MonotonicTopKPlan` with `must_consolidate: false`?

Reply:
It can happen when we have physical monotonicity, i.e., when what we lower from MIR comes with the `monotonic` flag set to `true`. This behavior change is actually an improvement brought about by this PR in that we will then not be keeping around an unnecessary arrangement when the input is append-only.